### PR TITLE
Tell emscripten to remove exception handling code when panic=abort

### LINKED ISSUE #36900: Tell emscripten to remove exception handling code when the panic runtime is used
emcc can remove invoke instructions during final translation. So when compiling with `-C panic-runtime=abort`, in order to strip the unwinding code out of std, we should pass the appropriate flags to emcc. This can probably be done be setting the `link_args` from the panic_abort crate.

### PR DESCRIPTION
**Pull request by pepyakin**
Fixes #36900

### CODE REVIEW

File: src/librustc_trans/back/link.rs [hunk 1]
```diff
@@ -712,6 +713,11 @@ fn link_natively(sess: &Session,
         cmd.arg(root.join(obj));
     }
 
+    if sess.target.target.options.is_like_emscripten &&
+       sess.panic_strategy() == PanicStrategy::Abort {
+        cmd.arg("-s DISABLE_EXCEPTION_CATCHING=1");
```
Comment (alexcrichton) [inline src/librustc_trans/back/link.rs]: I think this'll want to get passed as two separate arguments, right? (right now this is just being passed as one)

Comment (pepyakin) [inline src/librustc_trans/back/link.rs]: Yep, you are right. Will fix.

**Pull request merged**